### PR TITLE
Higher level alerts framework

### PR DESCRIPTION
**Pull request by eightyeight**
This introduces a new submodule called `alerts`. See the [readme](https://github.com/eightyeight/rtc-health/blob/higher-level-metrics/alerts/README.md) for more info.

### CODE REVIEW

**Comment by DamonOehlman** [comment]
Very, very nice Dan.  I'll leave @nathanoehlman look after merging this one :)

@nathanoehlman Are you able to add Dan to the list of npm users also for the package, it would be good to see his name against the list of people contributing to this one.  If not I can do it.

**Comment by eightyeight** [comment]
Happy to help :relaxed:

File: alerts/index.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+// rtc-health monitor module. That module provides low-level events that monitor
+// the rtc connections in great detail; this module provides high-level
+// user-friendly events that might be suitable for updating a UI.
+
+module.exports = function(monitor, opts) {
+	opts = opts || {};
+
+  if (!monitor) throw new Error('alerts must be given a monitor');
+
+	var alerter = new EventEmitter();
```
Comment (silviapfeiffer) [inline alerts/index.js]: indent

File: alerts/index.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      opts: opts,
+      context: context,
+      emit: emit,
+    });
+  }
+
+	// Listen for health reports on all qc connections and monitor the statistics
+	// we're interested in. This callback happens regularly, once for each peer
+	// connection.
+	monitor.on('health:report', function(reporter) {
```
Comment (silviapfeiffer) [inline alerts/index.js]: indent weird?

**Comment by silviapfeiffer** [comment]
good stuff! add yourself to https://github.com/rtc-io/CLA ?

**Comment by eightyeight** [comment]
I'll do an indentation pass soon, sorry about that :joy:

**Comment by eightyeight** [comment]
Okay this is getting closer but not quite there haha. I decided to go crazy and add a couple more types, but they have some issues.

**Comment by eightyeight** [comment]
Okay, it's all working for me now!

**Comment by DamonOehlman** [comment]
Yeah having a look at what Nathan has there is probably going to be a good start, but I'd suggest moving towards having a local switchboard instance used rather than relying on the dev switchboard instance.

Probably the best place to look beyond this package at how to create tests will be [`rtc-signaller`](https://github.com/rtc-io/rtc-signaller) as it still has it's own tests rather than standalone test packages which is taking it another degree further (great reusability, but more complex).

**Comment by nathanoehlman** [comment]
Hey Dan - nice PR, looks good!

Only problem I can see is problems with the debouncing if the `pollInterval` period is set to a greater or equal value to the threshold period, so I might just put a check in around that.

**Comment by nathanoehlman** [comment]
Also, with regards to testing, I might set this project up with a proper Travis CI build process, and move to using a local signaller as per @DamonOehlman's suggestion.

**Comment by eightyeight** [comment]
Good point catching the debouncing issue, that would definitely be a problem. I reckon at some point it would be good to replace the readme with tests, as long as the tests are fluent enough to convey the same information.

**Comment by nathanoehlman** [comment]
Published in `1.1.0`

**Pull request merged**